source/client_test.go

```diff
@@ -1,100 +1,38 @@
 package source
 
 import (
-	"math/rand"
 	"testing"
-	"time"
 
+	assert "github.com/stretchr/testify/require"
 	stripe "github.com/stripe/stripe-go"
-	"github.com/stripe/stripe-go/currency"
-	. "github.com/stripe/stripe-go/utils"
+	_ "github.com/stripe/stripe-go/testing"
 )
 
-func init() {
-	stripe.Key = GetTestKey()
-	rand.Seed(time.Now().UTC().UnixNano())
+func TestSourceGet(t *testing.T) {
+	source, err := Get("gold", nil)
+	assert.Nil(t, err)
+	assert.NotNil(t, source)
 }
 
-func TestSource(t *testing.T) {
-
-	sourceParams := &stripe.SourceObjectParams{
+func TestSourceNew(t *testing.T) {
+	source, err := New(&stripe.SourceObjectParams{
 		Type:     "bitcoin",
 		Amount:   1000,
-		Currency: currency.USD,
+		Currency: "USD",
```

Comment:
It's a little subtle, but I basically stopped using currency because it's a little bit of a failed abstraction. `stripe.Currency` is defined in the `stripe` package, but individual currencies are defined in the `currency` package, which needs to import `stripe` to get at `Currency`.

The problem with this is that it's impossible to use currencies from `stripe` and its tests! This really sucks, and probably currencies should have just gone into the top-level package like `stripe.USD`.

I basically stopped using them because you can't broadly apply them. That said, I don't really care that much. I added usage in 25b053f where it's possible.

Reply:
Gotcha, thanks for the detailed explanation. This explains why I haven't been able to reliably use it myself either because it was undefined half the time in my tests.

Reply:
No worries!